src/main/host/thread_ptrace.c

```diff
@@ -190,6 +247,55 @@ static Thread* _threadPtraceToThread(ThreadPtrace* thread) {
     return (Thread*)thread;
 }
 
+static const char* _regs_to_str(const struct user_regs_struct* regs) {
+    static char buf[1000];
+    ssize_t offset = 0;
+#define REG(x) offset += sprintf(&buf[offset], #x ":0x%llx ", regs->x);
+    REG(r15);
+    REG(r14);
+    REG(r15);
+    REG(r14);
+    REG(r13);
+    REG(r12);
+    REG(rbp);
+    REG(rbx);
+    REG(r11);
+    REG(r10);
+    REG(r9);
+    REG(r8);
+    REG(rax);
+    REG(rcx);
+    REG(rdx);
+    REG(rsi);
+    REG(rdi);
+    REG(orig_rax);
+    REG(rip);
+    REG(cs);
+    REG(eflags);
+    REG(rsp);
+    REG(ss);
+    REG(fs_base);
+    REG(gs_base);
+    REG(ds);
+    REG(es);
+    REG(fs);
+    REG(gs);
+#undef REG
+    return buf;
+}
+
+static const char* _syscall_regs_to_str(const struct user_regs_struct* regs) {
+    static char buf[1000];
+    ssize_t offset = 0;
+    sprintf(&buf[offset], "arg0:%lld ", regs->rdi);
+    sprintf(&buf[offset], "arg1:%lld ", regs->rsi);
+    sprintf(&buf[offset], "arg2:%lld ", regs->rdx);
+    sprintf(&buf[offset], "arg3:%lld ", regs->r10);
+    sprintf(&buf[offset], "arg4:%lld ", regs->r8);
+    sprintf(&buf[offset], "arg5:%lld", regs->r9);
+    return buf;
+}
```

Comment:
Should this be in a `#ifdef DEBUG` block, since it's only ever called from a `debug()` log macro (which is also wrapped in an `#ifdef DEBUG` block)?

Reply:
OK, cool :)